src/test/java/org/sagebionetworks/bridge/spring/controllers/EmailControllerTest.java

```diff
@@ -0,0 +1,140 @@
+import static org.sagebionetworks.bridge.TestUtils.mockEditAccount;
+import static org.testng.Assert.assertEquals;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.common.collect.Maps;
+
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import org.sagebionetworks.bridge.TestConstants;
+import org.sagebionetworks.bridge.config.BridgeConfig;
+import org.sagebionetworks.bridge.dao.AccountDao;
+import org.sagebionetworks.bridge.models.accounts.Account;
+import org.sagebionetworks.bridge.models.accounts.AccountId;
+import org.sagebionetworks.bridge.models.studies.Study;
+import org.sagebionetworks.bridge.services.StudyService;
+
+public class EmailControllerTest extends Mockito {
+
+    private static final String EMAIL = "email";
+    private static final String DATA_BRACKET_EMAIL = "data[email]";
+    private static final String HEALTH_CODE = "healthCode";
+    private static final String UNSUBSCRIBE_TOKEN = "unsubscribeToken";
+    private static final String TOKEN = "token";
+    private static final String STUDY2 = "study";
+    private static final String EMAIL_ADDRESS = "bridge-testing@sagebase.org";
+    private static final AccountId ACCOUNT_ID = AccountId.forEmail(TestConstants.TEST_STUDY_IDENTIFIER, EMAIL_ADDRESS);
+
+    @Mock
+    StudyService mockStudyService;
+
+    @Mock
+    AccountDao mockAccountDao;
+    
+    @Mock
+    Account mockAccount;
+    
+    @Mock
+    BridgeConfig mockConfig;
+    
+    @Mock
+    HttpServletRequest mockRequest;
+    
+    @Mock
+    HttpServletResponse mockResponse;
+    
+    @Spy
+    @InjectMocks
+    EmailController controller;
+
+    Study study;
+    
+    @BeforeMethod
+    public void before() {
+        MockitoAnnotations.initMocks(this);
+        when(mockConfig.getEmailUnsubscribeToken()).thenReturn(UNSUBSCRIBE_TOKEN);
+        when(mockAccountDao.getHealthCodeForAccount(ACCOUNT_ID)).thenReturn(HEALTH_CODE);
+        doReturn(mockRequest).when(controller).request();
+        doReturn(mockResponse).when(controller).response();
+        mockEditAccount(mockAccountDao, mockAccount);
+        
+        study = Study.create();
+        study.setIdentifier(TEST_STUDY_IDENTIFIER);
+        
+        when(mockStudyService.getStudy(TEST_STUDY_IDENTIFIER)).thenReturn(study);
+    }
+
+    private void mockContext(String... values) {
+        Map<String, String[]> paramMap = Maps.newHashMap();
+        for (int i = 0; i <= values.length - 2; i += 2) {
+            paramMap.put(values[i], new String[] { values[i + 1] });
+        }
+        for (Map.Entry<String, String[]> param : paramMap.entrySet()) {
+            when(mockRequest.getParameter(param.getKey())).thenReturn(param.getValue()[0]);
+        }
+    }
```

Comment:
In the old tests, we had separate tests for getting from query param or getting from the post body. Is this abstracted away in request.getParameter(). If not, do we need to separate these out again? What about the mixed() test?